### ISSUE TITLE
Variable Fonts: Check 'wght' axis default is 400, range is within 1-1000, matches legacy styles

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#1707]
We should have FB checks that begin by checking if it is a variable font, and if so, is it compliant with a few basic things. 

One is that fontmake allows users to make fonts with a 'wght' axis that isn't spec compliant, eg from eg 160-232. Spec says,

> Valid numeric range: Values must be in the range 1 to 1000.
>
> Scale interpretation: Values can be interpreted in direct comparison to values for usWeightClass in the 'OS/2' table, or the CSS font-weight property.
>
> Recommended or required “Regular” value: 400 is required.

- https://www.microsoft.com/typography/otspec/dvaraxistag_wght.htm

Should be easy to check

- [ ] default is 400
- [ ] range is within 1-1000
- [ ] if range is not starting/ending on a value cleanly divisible by 100, warn that it is unusual

**Comment by felipesanches** [comment]
CabinVFBeta from current master on `google/fonts` repo, has a defaultValue of 59.0 instead of 400. I think I can use it as a bad-reference for code-testing this check:

![screenshot at 2018-02-14 18 48 50](https://user-images.githubusercontent.com/213676/36227272-b71be724-11b7-11e8-8e6b-fe96015f9700.png)

**Comment by davelab6** [comment]
Sure

**Comment by felipesanches** [comment]
from these axis defined in the spec:

* 'ital' 	Italic
* 'opsz' 	Optical size
* 'slnt' 	Slant
* 'wdth' 	Width
* 'wght' 	Weight

which are strictly required in any varfont?

**Comment by felipesanches** [comment]
Did I get it right that all these are required (which these required default values in parentheses):
* 'ital' 	Italic (0 required)
* 'slnt' 	Slant (0 required)
* 'wdth' 	Width (100 required)
* 'wght' 	Weight (400 required)

And that only 'opsz' (Optical size) is not required, but, if available, then with a  dfaultValue between 9 and 13 ?

**Comment by felipesanches** [comment]
And the required axes would result in the check emitting a FAIL if any of them is missing.

**Comment by m4rc1e** [comment]
@felipesanches Not all of them are required. Some fonts may have just a wght axis, others just an ital axis.

Here's what I think we should test
- The origin should be the Regular weight (90% sure every font in GF which is MM compatible contains a regular)
- Weight axis range should be from 0-1000. When you export fonts from Glyphs it uses the 'Axes coordinates' field. In the past, designers have gotten the values for this field by measuring the stem of the 'H'

![screen shot 2018-02-15 at 07 53 52](https://user-images.githubusercontent.com/7525512/36245957-6a7bea22-1225-11e8-8e27-cade150edddb.png)

This means the range of the axis is very small e.g 32-240 (Like your CabinvVFBeta example), this is wrong. Designers need to set an additional field called Axis location for each master. We can then set the 400 value for the Regular master etc.

**Comment by felipesanches** [comment]
@davelab6 asked for:

* **(a)** "range is within 1-1000"
* **(b)** "if range is not starting/ending on a value cleanly divisible by 100, warn that it is unusual"

@m4rc1e suggested: "Weight axis range should be from 0-1000."

Should we expect that all fonts with a 'wght' have a range of exactly 0-1000 ? (If so, then testing **(b)** would not be necessary)

**Comment by moyogo** [comment]
The specs don't say the default should have a specific value. They specify
what value is required or recommended for the Regular instance, which
doesn't have to be the default instance.

**Comment by moyogo** [comment]
See https://docs.microsoft.com/en-gb/typography/opentype/spec/dvaraxisreg

For variable font implementations that support a given axis, the “Regular”
value will often be a good choice for the default value of that axis in the
'fvar' variation axis record. The default values set in the 'fvar' table,
however, are implementation-specific, and fonts are not required to use
this “Regular” value as the axis default value.

**Comment by moyogo** [comment]
You need to fix your checks. They should only enforce required or
recommended values for the Regular instance, not for the default when it's
not the Regular.

**Comment by davelab6** [comment]
Agree, thanks for the note Denis :)

**Comment by felipesanches** [comment]
So, all these new checks should be looking for the value of one of these axes **for the regular instance** (instead of the default axis value) and then FAIL or WARN if they're not the recommended or required values, right?

I'll take care of this, then.

**Comment by davelab6** [comment]
Yes the regular at 400 is a fail, the bold at 700 too I think, and the rest
can be warnings

**Comment by felipesanches** [comment]
Cool.

- [ ] FAIL when regular is not 400 
- [ ] FAIL when bold is not 700

**Comment by felipesanches** [comment]
All Regular:axis and Bold:wght checks have been fixed above (thanks @moyogo!)

I still need feedback on https://github.com/googlefonts/fontbakery/issues/1707#issuecomment-366029812

**Comment by arrowtype** [comment]
> "Weight axis range should be from 0-1000."
> 
> Should we expect that all fonts with a 'wght' axis have a range of exactly 0-1000 ? (If so, then testing (b) would not be necessary)

No, it should be 1–1000. From the [wght spec](https://docs.microsoft.com/en-us/typography/opentype/spec/dvaraxistag_wght):

> Valid numeric range: Values must be in the range 1 to 1000.

**Comment by arrowtype** [comment]
Also, it is not required that "wght default is 400". The spec says:

> Recommended or required “Regular” value: 400 is required.

Variable fonts should probably include a `400` value for regular, but this needn’t be the _default_ value of the wght axis.

**Comment by RosaWagner** [comment]
The Open Type Spec requires "Italic" (instead of "Regular Italic") in Name ID2 to make sure style linking will work correctly. Then it is recommended to not generate name IDs 16 and 17 for the RIBBI styles since they would be the same as name IDs 1 and 2, but if you want "Regular Italic" it should be like this:

Name ID 1: Family Name
Name ID 2: Italic
Name ID 16: Family Name
Name ID 17: Regular Italic

This scheme is unconventional and against opentype recommendations, therefore Fontbakery will continue to FAIL this check (at least for googlefonts), but from experience, it's okay the font will work everywhere.

Additional info:
https://learn.microsoft.com/en-us/typography/opentype/spec/name#name-ids
